### PR TITLE
fix(form/TFD-1785): provide a password renderer

### PR DESCRIPTION
**Pull request by sgendre**
**What is the problem this PR is trying to solve?**
Password are displayed on forms with read-only mode
image

**What is the chosen solution to this problem?**
Replace characters by bullets
image

**Please check if the PR fulfills these requirements**
- [x] The PR commit message follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md)
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] Related design / discussions / pages (not in jira), if any, are all linked or available in the PR

**[ ] This PR introduces a breaking change**

### CODE REVIEW

File: packages/components/src/JSONSchemaRenderer/JSONSchemaRenderer.component.js [hunk 1]
```diff
@@ -72,6 +72,20 @@ function TextRenderer({ propertyKey, title, properties }) {
 	);
 }
 
+function PasswordRenderer({ propertyKey, title, properties }) {
```
Comment (acateland) [inline packages/components/src/JSONSchemaRenderer/JSONSchemaRenderer.component.js]: please export and test

File: packages/components/src/JSONSchemaRenderer/JSONSchemaRenderer.component.js [hunk 2]
```diff
@@ -146,6 +164,17 @@ function typeResolver(schema, uiSchema) {
 		const type = schema[e[0]].type;
 		const title = schema[e[0]].title;
 
+		if (isPassword(uiSchema, e[0])) {
```
Comment (acateland) [inline packages/components/src/JSONSchemaRenderer/JSONSchemaRenderer.component.js]: please add a test for this behavior

Comment (sgendre) [reply packages/components/src/JSONSchemaRenderer/JSONSchemaRenderer.component.js]: done

File: packages/components/src/JSONSchemaRenderer/JSONSchemaRenderer.component.js (hunk 1 above)
Comment (sgendre) [reply packages/components/src/JSONSchemaRenderer/JSONSchemaRenderer.component.js]: tested through JSONSchemaRenderer on line 133

File: packages/components/src/JSONSchemaRenderer/JSONSchemaRenderer.component.js [hunk 3]
```diff
@@ -72,6 +73,20 @@ function TextRenderer({ propertyKey, title, properties }) {
 	);
 }
 
+export function PasswordRenderer({ propertyKey, title, properties }) {
+	return (
+		<div className={classNames('text-renderer', `text-renderer-${propertyKey}`)} key={propertyKey}>
+			<dt>{title || propertyKey}</dt>
+			<dd>{'\u2022'.repeat(properties ? properties.length : 0)}</dd>
```
Comment (jmfrancois) [inline packages/components/src/JSONSchemaRenderer/JSONSchemaRenderer.component.js]: I would do '\u2022'.repeat(6)

**Pull request merged**